homeassistant/components/knx/storage/entity_store_validation.py

```diff
@@ -0,0 +1,69 @@
+"""KNX Entity Store Validation."""
+
+from typing import Literal, TypedDict
+
+import voluptuous as vol
+
+from .entity_store_schema import ENTITY_STORE_DATA_SCHEMA
+
+
+class _ErrorDescription(TypedDict):
+    path: list[str] | None
+    error_message: str
+    error_class: str
+
+
+class EntityStoreValidationError(TypedDict):
+    """Negative entity store validation result."""
+
+    success: Literal[False]
+    error_base: str
+    errors: list[_ErrorDescription]
+
+
+class EntityStoreValidationSuccess(TypedDict):
+    """Positive entity store validation result."""
+
+    success: Literal[True]
+    entity_id: str | None
+
+
+def parse_invalid(exc: vol.Invalid) -> _ErrorDescription:
+    """Parse a vol.Invalid exception."""
+    return _ErrorDescription(
+        path=[str(path) for path in exc.path],  # exc.path: str | vol.Required
+        error_message=exc.msg,
+        error_class=type(exc).__name__,
+    )
+
+
+def validate_entity_data(entity_data: dict) -> dict:
+    """Validate entity data. Return validated data or raise EntityStoreValidationException."""
```

Comment:
Please follow PEP8 for docstrings. Lines should be max 72 characters. The first line is a header and should be a single sentence. After that an optional body may follow separated from the header by a blank line.

Reply:
👍 added fix to https://github.com/home-assistant/core/pull/125734